firebase-app-distribution/src/main/java/com/google/firebase/appdistribution/UpdateAppClient.java

```diff
@@ -79,10 +109,185 @@ private void redirectToPlayForAabUpdate(String downloadUrl, Activity currentActi
     this.updateTask.updateProgress(updateState);
   }
 
+  public void updateApk(String downloadUrl, Activity currentActivity) {
+    getDownloadTask(downloadUrl)
+        .addOnSuccessListener(
+            file -> {
+              install(file.getPath(), currentActivity)
+                  .addOnSuccessListener(
+                      Void -> {
+                        UpdateState updateState =
+                            UpdateState.builder()
+                                .setApkTotalBytesToDownload(0)
+                                .setApkBytesDownloaded(file.length())
+                                .setUpdateStatus(UpdateStatus.INSTALLED)
+                                .build();
+                        updateTask.updateProgress(updateState);
+                        updateAppTaskCompletionSource.setResult(updateState);
+                      })
+                  .addOnFailureListener(
+                      e ->
+                          setUpdateAppTaskCompletionError(
```

Comment:
This will mask all errors during install and always return INSTALLATION_FAILURE. Might be worth just setting it to `e`

Reply:
changed it to use e if its firebase exception and install failure as default similar to updatetolatestrelease